### ISSUE TITLE
Impossible to upload some themes

### ISSUE THREAD

**Issue by doclucas** [thirtybees/thirtybees#579]
As discussed here:
https://forum.thirtybees.com/topic/1969/impossible-to-upload-any-theme

The fix is:
Edit controllers/admin/AdminThemesController.php
**Change:**
`$xmlFile = $sandbox.$themeDir.'/config.xml';`
to
`$xmlFile = $sandbox.$themeDir.'/Config.xml';`

Or better if you can modify it to accept both `config.xml` and `Config.xml`, as I believe there are some themes that may incorrectly be packaged with `config.xml`

**Comment by Traumflug** [comment]
Related commits:

https://github.com/thirtybees/thirtybees/commit/48521e43c18ba4da2071655b8aa5b5191ab73954
https://github.com/thirtybees/thirtybees/commit/8e45e1ee49d69a70fa190812f246e4d805ac2014

**Comment by Traumflug** [comment]
Looks like this needs some refactoring.

- There is `Themes->getConfiguration()` which deal with both cases, but isn't used in *AdminThemesController*.
- `Themes->getConfiguration()` does a `json_encode()`/`json_decode()` cycle, which appears to be a waste.
- *AdminThemesController* comes with `checkXmlFields()`, which should belong to class *Themes* and probably go into `Themes->getConfiguration()`.

**Comment by getdatakick** [comment]
I have prepared a fix for this issue, see commit c74520426237ca8b095b05231ac092c149185f1d. @Traumflug please review. I'm sure there will be many questions :)

### Some background:

Themes are distributed in zip files with following content:

```
modules
  +- moduleA
  +- moduleB
themes
  +- theme-name
    +- 404.tpl
    +- address.tpl
    +- product.tpl
    +- config.xml
    +- ...
Config.xml
```
Zip file can contains multiple modules, many variants of theme, docs,.... And it must contain `Config.xml` (with capital C) that describes how to install these *themes*

When theme is installed, the `Config.xml` file is copied to `/config/xml/themes/theme-name.xml`, modules are copied to `/modules` directory, and the theme variants to `/themes` directory. 

Note that there exists another `config.xml` file in `theme-name` directory, this time with lowercase c. This config file has nothing to do with `Config.xml` file -- it is remnant of old ps14 technology. This file was used to advertise theme capabilities. 

Example:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<theme>
	<version value="1.4">
		<ccc available="true" />
		<guest_checkout available="true" />
		<one_page_checkout available="true" />
		<store_locator available="true" />
	</version>
</theme
```
Compare with [Config.xml](https://github.com/thirtybees/thirtybees/blob/1.0.x/config/xml/themes/default.xml) file from installation package.

Code that checks these capabilities in `config.xml` no longer works, and it was not called anywhere. It actually wasn't even executable, so I've removed id from the codebase in this changelist.

Another note - just to make things a bit more complicated, `community-theme-default` theme contains [config.xml](https://github.com/thirtybees/community-theme-default/blob/1.0.x/config.xml) file with a format of `Config.xml` file. 

To make things even more complicated, it's possible that the theme is *unpacked* in the servers `/themes` directory, but not installed. In that case, there exists directory with the same format as the zip package, with `Config.xml`. Thirtybees will offer installation of this theme. When you perform this installation, this directory in replaced in-situ with the variant theme from inside this directory. 

```
/themes
  +- community-theme-default
    +- 404.tpl
    +- address.tpl
    +- product.tpl
    +- config.xml
    +- ...
  +- notinstalled
    +- modules
      +- moduleA
      +- moduleB
    +- themes
      +- notinstalled
        +- 404.tpl
        +- address.tpl
        +- product.tpl
        +- config.xml
        +- ...
    +- Config.xml
  +- installed
    +- 404.tpl
    +- address.tpl
    +- product.tpl
    +- config.xml
    +- ...
```

### Takeaways

- if theme is installed, we have to look for it's config file in `/config/xml/themes/<theme-name>.xml`
- if theme is not installed, but it is upload to server, the config file is in `/themes/<theme-name>/Config.xml`
- disregard `config.xml` file for installed themes

If you have read this far, take an aspirin.

**Comment by getdatakick** [comment]
Closed by 89964b6def8c0eba43d50ee71ca0e984ab1667f9
